app/models/spree/calculator/avalara_transaction.rb

```diff
@@ -54,6 +56,8 @@ def can_calculate_tax?(order, item)
     end
 
     def get_avalara_response(order)
+      Rails.cache.delete(cache_key(order)) if order.force_tax_recalculation
```

Comment:
this cache avoidance is pretty dumb and can be improved with adding the second cache_key with shorter TTL for recalculation task